.misc/build_and_test.sh

```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+# this is helper script for travis
+# it checks if we should run tests
```

Comment:
not really, it actually executes also all test steps in the right cases. If you want to mention this, do it directly above the relevant if-clause :)